src/components/ChooseCheckType.test.tsx

```diff
@@ -0,0 +1,91 @@
+import React from 'react';
+import { config } from '@grafana/runtime';
+import { screen } from '@testing-library/react';
+import { apiRoute } from 'test/handlers';
+import { render } from 'test/render';
+import { server } from 'test/server';
+
+import { FeatureName } from 'types';
+
+import { ChooseCheckType } from './ChooseCheckType';
+
+function renderChooseCheckType(checkLimit = 10, scriptedLimit = 10) {
+  server.use(
+    apiRoute('getTenantLimits', {
+      result: () => ({
+        json: {
+          MaxChecks: checkLimit,
+          MaxScriptedChecks: scriptedLimit,
+          MaxMetricLabels: 16,
+          MaxLogLabels: 13,
+          maxAllowedMetricLabels: 10,
+          maxAllowedLogLabels: 5,
+        },
+      }),
+    })
+  );
+  return render(<ChooseCheckType />);
+}
+it('shows check type options with scripted feature off', async () => {
+  renderChooseCheckType();
+  const checkTypes = ['HTTP', 'TCP', 'DNS', 'PING', 'MULTIHTTP', 'Traceroute'];
+  const cards = await Promise.all(
+    checkTypes.map((checkType) => {
+      return screen.findByText(checkType);
+    })
+  );
+  cards.forEach((card) => {
+    expect(card).toBeInTheDocument();
+  });
+});
+
+it('shows check type options with scripted feature on', async () => {
+  // @ts-ignore
+  config.featureToggles[FeatureName.ScriptedChecks] = true;
+
+  renderChooseCheckType();
+  const checkTypes = ['HTTP', 'TCP', 'DNS', 'PING', 'MULTIHTTP', 'Traceroute', 'Scripted'];
+  const cards = await Promise.all(
+    checkTypes.map((checkType) => {
+      return screen.findByText(checkType);
+    })
+  );
+  cards.forEach((card) => {
+    expect(card).toBeInTheDocument();
+  });
+});
+
+it('shows error alert when check limit is reached', async () => {
+  renderChooseCheckType(1);
+  const errorAlert = await screen.findByText('Check limit reached');
+  expect(errorAlert).toBeInTheDocument();
+  const checkTypes = ['HTTP', 'TCP', 'DNS', 'PING', 'MULTIHTTP', 'Traceroute', 'Scripted'];
+  const cards = await Promise.all(
+    checkTypes.map((checkType) => {
+      return screen.queryByText(checkType);
+    })
+  );
+  cards.forEach((card) => {
+    expect(card).not.toBeInTheDocument();
+  });
+});
+
+it('shows error alert when scripted check limit is reached', async () => {
+  // @ts-ignore
+  config.featureToggles[FeatureName.ScriptedChecks] = true;
```

Comment:
We could do, I don't have feelings on hiding the card vs restyling the card and making it disabled. I originally thought I'd just disable the card, but we don't have great design patterns for disabled cards actually looking disabled (in grafana ui they just turn off the pointer events and make no visual distinction, which is just confusing imo).

I think that the test passing with the flag off is still kinda valid? Like I'm now sure _how_ a user who doesn't have access to scripted checks in the UI would hit the limits, but I don't see why we'd want the experience to be different in that scenario. Like if they were using terraform or something and the flag states were out of sync with each other, I don't think it's bad to notify the user they're at their limit.

Reply:
>We could do, I don't have feelings on hiding the card vs restyling the card and making it disabled. I originally thought I'd just disable the card, but we don't have great design patterns for disabled cards actually looking disabled (in grafana ui they just turn off the pointer events and make no visual distinction, which is just confusing imo).

Yeah, fair point. 👍 

>I think that the test passing with the flag off is still kinda valid?

It's just a bit of a red herring. The thing is the tests double as documentation so if I were to read this in a few months' time and notice the feature toggle had been explicitly set in the test but we don't account for that in the code I'd think our implementation is wrong and might fix it. I'd prefer this test if we either explicitly set the feature toggle to false with a comment or we just removed it altogether as it has no affect.

I don't know if this would ever be a thing but I was wondering if we would ever set their limit to 0 scripted checks then they would see this regardless of if they had access or knew scripted checks were a thing.

Also, we want to remove `config.featureToggles[FeatureName.ScriptedChecks] = true;` this line in this test too as it is still going to pollute other tests :)